src/main/java/org/apache/ibatis/io/DefaultVFS.java

```diff
@@ -95,24 +95,24 @@ public boolean isValid() {
              * then we assume the current resource is not a directory.
              */
             is = url.openStream();
-            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-            List<String> lines = new ArrayList<>();
-            for (String line; (line = reader.readLine()) != null;) {
-              if (log.isDebugEnabled()) {
-                log.debug("Reader entry: " + line);
-              }
-              lines.add(line);
-              if (getResources(path + "/" + line).isEmpty()) {
-                lines.clear();
-                break;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
```

Comment:
Since you are using `BufferedReader` I would reset it at the end to ensure there is no memory being allocated to it after it is done being used.
I think you have to use `.mark()` before you can use `.reset()`. Not entirely sure

Reply:
The point of e `try` statement added is to automatically close the `BufferedReader` after the `try` ends. See [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

Reply:
Ok then. I didn't know that.. as most of my try blocks were for assert statements and a few users asked me about data  clearance :)

Reply:
Well, the idea of using `try-with-resources` statement is that `close` method is called after leaving the block and that should release system resources.